web/vview/widgets/tag-search-dropdown.js

```diff
@@ -1032,6 +1032,83 @@ class TagSearchDropdownWidget extends widget
         return false;
     }
 
+    // Composing tag groups by matching translation in lowercase with brackets stripped out.
+    // joinMonotags provides split and mono tag handling e.g. handglove like hand_glove, hand-glove, hand glove will reside in hand group
+    // joinMonotags strictly applyed only to tags starting in lowercase since it's likely not name of the character
+    // joinPrefixes will try to join formed groups with same prefix
+    _groupTagsByTranslation = (autocompletedTags, translatedTags, options) => {
+        const tagGroupReducer = (acc, tag) => {
+            const strippedTag = tag.tag.replace(/\s*\(.+\)\s*/g, '');
+
+            // Consider translated itself if defined as property but does not have a value
+            if (!Object.hasOwn(translatedTags, strippedTag)) {
+                acc.standalone.push(tag);
+                return acc;
+            }
+
+            const translated = translatedTags[strippedTag] ?? tag.tag;
+            let slug = translated.toLowerCase();
+
+            if(options.joinMonotags) {
+                // Likely not name since starting with lowercase
+                if (translated[0] === slug[0]) {
+                    // Attach to group if starts with any existing group name for monotags and tags with spaces handling
+                    slug = Object.keys(acc.groups).find(key => slug.startsWith(key)) ?? slug;
+                    slug = slug.split(/[ _-]/g)[0]
+                }
+            }
+
+            if (!acc.groups[slug]) {
+                acc.groups[slug] = {
+                    tag: new Set([tag.tag]),
+                    // Downside of this approach is that joined tag list shoud be inserted in fixed place since we dont know position
+                    search: tag.search.replace(tag.tag, '')
```

Comment:
Probably need to reparse  boundaries and store it for later insert position inference will fix later